src/lib/stores.js

```diff
@@ -0,0 +1,32 @@
+import { getContext, setContext } from 'svelte';
+import { writable } from 'svelte/store';
+import { browser } from '$app/environment';
+
+// safely create a store in order to avoid
+// https://github.com/sveltejs/kit/discussions/4339
+
+const KEY = 'realworld-stores';
+
+const createStores = () => ({
+	user: writable()
+});
+
+let globalStores;
+export const initStores = () => {
+	if (browser) {
+		globalStores = createStores();
+	} else {
+		setContext(KEY, createStores());
+	}
+}
+
+const getStores = () => browser ? globalStores : getContext(KEY);
+
+export const user = {
+	subscribe(fn) {
+		return getStores().user.subscribe(fn);
+	},
+	set(value) {
+		return getStores().user.set(value);
+	}
+};
```

Comment:
This doesn't feel like the sort of code we want to be encouraging either. It's very confusing.

We're already exposing the user via data from `+layout.server.js`. We should just access the user everywhere via `$page.data.user`, not with this special store.

Reply:
The first commit in this PR used `data.user` everywhere, but it was a bit problematic because the only way to update it was to re-request all data. So if you want to do something like update the user display name you have to make one request to set it and then call `invalidateAll` to re-request all data eventhough we already have all the necessary data.

We had talked at one point about exposing SvelteKit's version of this so that users don't have to write special store handling code themselves. I think that would be a nice thing to do. Even if we don't use a store here, users do need to jump through some hoops if they ever want to use a store in their application for any other purpose

Reply:
Another solution would be to make `$page.data` writable. If it were, then I could simply update it instead of needing to create a new store